packages/react/src/internals/context.tsx

```diff
@@ -105,22 +105,26 @@ export type TRPCContextState<
     pathAndInput: [TPath, TInput?],
     options?: InvalidateOptions,
   ) => Promise<void>;
+
   /**
    * @link https://react-query.tanstack.com/guides/query-invalidation
    */
-  invalidateQueries:
-    | (<
-        TPath extends keyof TRouter['_def']['queries'] & string,
-        TInput extends inferProcedureInput<TRouter['_def']['queries'][TPath]>,
-      >(
-        pathAndInput: [TPath, TInput?] | TPath,
-        filters?: InvalidateQueryFilters,
-        options?: InvalidateOptions,
-      ) => Promise<void>)
-    | ((
-        filters?: InvalidateQueryFilters,
-        options?: InvalidateOptions,
-      ) => Promise<void>);
+  invalidateQueries<
+    TPath extends keyof TRouter['_def']['queries'] & string,
+    TInput extends inferProcedureInput<TRouter['_def']['queries'][TPath]>,
+  >(
+    pathAndInput?: [TPath, TInput?] | TPath,
+    filters?: InvalidateQueryFilters,
+    options?: InvalidateOptions,
+  ): Promise<void>;
+  /**
+   * @link https://react-query.tanstack.com/guides/query-invalidation
+   */
+  invalidateQueries(
+    filters?: InvalidateQueryFilters,
```

Comment:
This `InvalidateQueryFilters` takes a genric but I didn't bother typing it for now, feel free to do a PR with that.. it'll be quite tricky